README.md

```diff
@@ -70,6 +68,38 @@ libsecp256k1 is built using autotools:
 
 To compile optional modules (such as Schnorr signatures), you need to run `./configure` with additional flags (such as `--enable-module-schnorrsig`). Run `./configure --help` to see the full list of available flags.
 
+Building with CMake
+-------------------
+
+To maintain a pristine source tree, CMake encourages to perform an out-of-source build by using a separate dedicated build tree.
+
+    $ rm -rf build && mkdir build && cd build
+    $ cmake ..
+    $ make
+    $ sudo make install  # optional
+
+To adjust the build system configuration, the following options can be provided to `cmake` call:
+
+| Option | Default value | Description |
```

Comment:
Does cmake have the equivalent of `./configure --help` that will just output the options? Otherwise you are creating documentation that will by default, go out of date, and/or need to be manually updated to stay in sync with the build system. It's easier to just get the options from the build system itself (as above), then they will always be correct.

Reply:
> Does cmake have the equivalent of `./configure --help` that will just output the options?

```sh
make -LH -N ..
```